finetune/model.py

```diff
@@ -118,7 +118,8 @@ def _model_fn(features, labels, mode, params):
 
         if mode == tf.estimator.ModeKeys.TRAIN:
             total_num_steps = params.n_epochs * params.dataset_size//params.batch_size
-            lr_decay = lambda lr, global_step: lr * schedules[params.lr_schedule](tf.to_float(global_step) / total_num_steps)
+            #CHANGE: LR, GLOBAL STEP, TOTAL STEPS IS NOW DIRECTLY TURNED IN TO SCHEDULE FUNCTION
+            lr_decay = lambda lr, global_step: lr * schedules[params.lr_schedule](tf.to_float(global_step) / total_num_steps,lr,tf.to_float(global_step),total_num_steps)
```

Comment:
I would remove this from this PR and move the stuff related to cyclic learning rates to a different branch so we can get this merged in.